src/clients/__mocks__/keycloak-mock.ts

```diff
@@ -11,65 +11,86 @@ export const mockMutatorGetter = (): MockMutator => {
-        setTimeout((): void => {
-          // eslint-disable-next-line no-unused-expressions
-          loadProfileRejectPayload
-            ? reject(loadProfileRejectPayload)
-            : resolve(loadProfileResolvePayload);
-        }, mockMutator.promiseTimeout);
-      }) as Keycloak.KeycloakPromise<Keycloak.KeycloakProfile, void>;
-    },
-    get tokenParsed(): Record<string, unknown> {
-      return mockMutator.getTokenParsed();
-    },
-    get token(): string | undefined {
-      return mockMutator.getTokens().token;
-    },
-    get idToken(): string | undefined {
-      return mockMutator.getTokens().idToken;
-    },
-    get refreshToken(): string | undefined {
-      return mockMutator.getTokens().refreshToken;
+  const partialMock = {
+    ...kc,
+    ...{
+      init(
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        initOptions?: Keycloak.KeycloakInitOptions
+      ): Keycloak.KeycloakPromise<boolean, Keycloak.KeycloakError> {
+        mockMutator.initCalled();
+        const clientInitRejectPayload = mockMutator.getClientInitRejectPayload();
+        const clientInitResolvePayload = mockMutator.getClientInitResolvePayload();
+        return new Promise((resolve: Function, reject: Function) => {
+          setTimeout((): void => {
+            // eslint-disable-next-line no-unused-expressions
+            clientInitRejectPayload
+              ? reject(clientInitRejectPayload)
+              : resolve(clientInitResolvePayload);
+          }, mockMutator.promiseTimeout);
+        }) as Keycloak.KeycloakPromise<boolean, Keycloak.KeycloakError>;
+      },
+      login: (
+        options?: Keycloak.KeycloakLoginOptions | undefined
+      ): Keycloak.KeycloakPromise<void, void> => {
+        mockMutator.loginCalled(options);
+        return Promise.resolve() as Keycloak.KeycloakPromise<void, void>;
+      },
+      logout: (
+        options?: Keycloak.KeycloakLogoutOptions | undefined
+      ): Keycloak.KeycloakPromise<void, void> => {
+        mockMutator.logoutCalled(options);
+        mockMutator.setUser();
+        return Promise.resolve() as Keycloak.KeycloakPromise<void, void>;
+      },
+      loadUserProfile(): Keycloak.KeycloakPromise<
+        Keycloak.KeycloakProfile,
+        void
+      > {
+        const loadProfileRejectPayload = mockMutator.getLoadProfileRejectPayload();
+        const loadProfileResolvePayload = mockMutator.getLoadProfileResolvePayload();
+        return new Promise((resolve: Function, reject: Function) => {
+          setTimeout((): void => {
+            // eslint-disable-next-line no-unused-expressions
+            loadProfileRejectPayload
+              ? reject(loadProfileRejectPayload)
+              : resolve(loadProfileResolvePayload);
+          }, mockMutator.promiseTimeout);
+        }) as Keycloak.KeycloakPromise<Keycloak.KeycloakProfile, void>;
+      }
     }
   };
+
+  const defineMockProperty = (name: string): void => {
```

Comment:
Getters and setters must be added after objects have been merged with spread operator {...keycloak, ...other}